lib/util/http/retry-after.spec.ts

```diff
@@ -0,0 +1,150 @@
+import { RequestError } from 'got';
+import * as hostRules from '../host-rules';
+import {
+  extractRetryAfterHeaderSeconds,
+  getMaxRetryAfter,
+  wrapWithRetry,
+} from './retry-after';
+
+function requestError(
+  response: {
+    statusCode?: number;
+    headers?: Record<string, string | string[]>;
+  } | null = null,
+) {
+  const err = new RequestError('request error', {}, null as never);
+  if (response) {
+    (err as any).response = response;
```

Comment:
why not passing as constructor parameter?

Reply:
It's complicated class, there is no straightforward way to construct it

Reply:
can't our partial helper work?